packages/members-ssr/index.js

```diff
@@ -203,6 +215,18 @@ class MembersSSR {
         const token = Array.isArray(query.token) ? query.token[0] : query.token;
         const member = await this._getMemberDataFromToken(token);
 
+        // perform and store geoip lookup for members when they log in
+        // TODO: is it preferable to do this here or in members-ssr consumer?
```

Comment:
`members-ssr` (maybe this part especially) feels almost like it's blurring the bounds a bit between library and app-level concerns. Does it make sense to keep this here, or would it make more sense for the consumer (Ghost) to be in control of the flow by orchestrating between `members-ssr` and `members-api`?

Reply:
Yeah I totally see where you're coming from here - I _think_ this is okay - and maybe you will too, if I give you some background / my "concept" of this module.

The reason we made this "members-ssr" as a separate thing was because I was trying to think of a future where people were not necessarily using the theme layer/frontend - but a custom frontend (like Gatsby or whatever)

This library _should_ be able to be dropped into any node app, and as long as the "magic link" points to that app - they can create a session with the app and do everything they need to.

This would obviously depend on them being able to pass in a `getMembersApi` fn - but this could easily be a proxy to their Ghost Admin's members api.

Because the geolocation stuff seems to be a core part of members, it would make sense to me that we would want this functionality to still work on a custom frontend - and so it makes sense to me, for it to be here.